contracts/finance/andromeda-mint-burn/src/testing/tests.rs

```diff
@@ -0,0 +1,495 @@
+            deposits: HashMap::new(),
+        },
+        ResourceRequirement {
+            resource: Resource::Cw20Token {
+                cw20_addr: AndrAddr::from_string(MOCK_CW20_CONTRACT_2.to_string()),
+            },
+            amount: Uint128::new(20000),
+            deposits: HashMap::new(),
+        },
+    ];
+
+    let output = Resource::Nft {
+        cw721_addr: AndrAddr::from_string(MOCK_NFT_CONTRACT_TO_MINT.to_string()),
+        token_id: "test_nft_to_mint_1".to_string(),
+    };
+
+    let sender = info.sender.as_str();
+
+    create_order(deps.as_mut(), requirements.clone(), output, sender).unwrap();
+
+    let created_order: GetOrderInfoResponse =
+        query_order_info(deps.as_ref(), Uint128::one()).unwrap();
+    assert_eq!(created_order.requirements, requirements.clone());
+
+    let hook_msg = Cw20HookMsg::FillOrder {
+        order_id: Uint128::one(),
+    };
+    let cw20_receive_msg = Cw20ReceiveMsg {
+        sender: "origin_cw20_sender".to_string(),
+        amount: Uint128::new(5000),
+        msg: encode_binary(&hook_msg).unwrap(),
+    };
+
+    receive_cw20(deps.as_mut(), cw20_receive_msg, MOCK_CW20_CONTRACT).unwrap();
+
+    let cw20_receive_msg = Cw20ReceiveMsg {
+        sender: "origin_cw20_sender_2".to_string(),
+        amount: Uint128::new(6000),
+        msg: encode_binary(&hook_msg).unwrap(),
+    };
+
+    receive_cw20(deps.as_mut(), cw20_receive_msg, MOCK_CW20_CONTRACT).unwrap();
+
+    let created_order: GetOrderInfoResponse =
+        query_order_info(deps.as_ref(), Uint128::one()).unwrap();
+
+    let order_status = created_order.order_status;
+    assert_eq!(order_status, OrderStatus::NotCompleted);
+
+    let cw20_receive_msg = Cw20ReceiveMsg {
+        sender: "origin_cw20_sender".to_string(),
+        amount: Uint128::new(7000),
+        msg: encode_binary(&hook_msg).unwrap(),
+    };
+
+    receive_cw20(deps.as_mut(), cw20_receive_msg, MOCK_CW20_CONTRACT).unwrap();
+
+    let cw20_receive_msg = Cw20ReceiveMsg {
+        sender: "origin_cw20_sender_2".to_string(),
+        amount: Uint128::new(4000),
+        msg: encode_binary(&hook_msg).unwrap(),
+    };
+
+    receive_cw20(deps.as_mut(), cw20_receive_msg, MOCK_CW20_CONTRACT).unwrap();
+
+    let created_order: GetOrderInfoResponse =
+        query_order_info(deps.as_ref(), Uint128::one()).unwrap();
+
+    let order_status = created_order.order_status;
+    assert_eq!(order_status, OrderStatus::NotCompleted);
+
+    let cw20_receive_msg = Cw20ReceiveMsg {
+        sender: "origin_cw20_sender".to_string(),
+        amount: Uint128::new(24000),
+        msg: encode_binary(&hook_msg).unwrap(),
+    };
+
+    receive_cw20(deps.as_mut(), cw20_receive_msg, MOCK_CW20_CONTRACT_2).unwrap();
+
+    let created_order: GetOrderInfoResponse =
+        query_order_info(deps.as_ref(), Uint128::one()).unwrap();
+
+    let order_status = created_order.order_status;
+    assert_eq!(order_status, OrderStatus::Completed);
```

Comment:
Is there any reason this can't be done in a loop? Seems to make the test unnecessarily long